src/Illuminate/Support/Collection.php

```diff
@@ -768,6 +768,16 @@ public function toJson($options = 0)
 	}
 
 	/**
+	 * Get an hash of the collection.
+	 *
+	 * @return string
+	 */
+	public function hash()
+	{
+		return md5($this);
```

Comment:
I agree with @ibrasho. An `equals()` methods makes more sense for the given use case.

Regarding hashing function: I think MD5 would be just fine for this use-case, since it's faster. And the chance to **randomly** encounter a hash collision is negligible,